app/Http/Controllers/AgreementController.php

```diff
@@ -5,34 +5,52 @@
 use Illuminate\Http\Request;
 use Illuminate\Support\Facades\Auth;
 use App\Agreement;
+use GrahamCampbell\Markdown\Facades\Markdown;
+use Arr;
 
 class AgreementController extends Controller
 {
 
-    private function get_agreement(){
-        $tos = Agreement::where("in_effect_at", "<=", now())->latest()->first();
+    public function view() {
 
-        return [
-            'title' => "Terms of Service",
-            'date' => sprintf("Last updated: %s", $tos->in_effect_at->format('jS \\of F, Y')),
-            'markdown' => $tos->content
-        ];
-    }
+        $tos = Agreement::current();
 
-    public function view() {
-        return view('pages.markdown', $this->get_agreement());
+        return view('pages.markdown',
+            [
+                'title' => "Terms and Conditions",
+                'date' => sprintf("Effective date: %s", $tos->in_effect_at->format('jS \\of F, Y')),
+                'version' => $tos->id,
+                'markdown' => Markdown::convertToHtml($tos->content)
+            ]
+        );
     }
 
     public function show(Request $request) {
-        return view('pages.prompt-tos', array_merge(
-            $this->get_agreement(),
-            ['intended' => $request->input('intended') ?? 'calendars']
-        ));
+
+        $tos = Agreement::current();
+
+        return view('pages.prompt-tos',
+            [
+                'title' => "Terms and Conditions",
+                'date' => sprintf("Effective date: %s", $tos->in_effect_at->format('jS \\of F, Y')),
+                'version' => $tos->id,
+                'markdown' => Markdown::convertToHtml($tos->content),
+                'intended' => $request->input('intended') ?? 'calendars'
+            ]
+        );
     }
 
-    public function agreement_accepted(Request $request) {
+    public function accept(Request $request) {
+        if(!Arr::has($request->all(), 'policy_acceptance')){
+            return redirect('pages.prompt-tos');
+        }
+
         Auth::user()->acceptAgreement();
 
+        if(Arr::has($request->all(), 'marketing_acceptance')){
+            Auth::user()->setMarketingStatus(true);     // We're not passing false here, as we're not opting out
```

Comment:
Maybe worth setting a default value of `true` so that we can just `setMarketingStatus()`?